### PR TITLE
Refactoring to make testing easier & adding tests

### PR DESCRIPTION
**Pull request by syou6162**
## Why
- ref: https://github.com/z3z1ma/dbt-osmosis/issues/93

## What
@z3z1ma 

- To adding tests easier, I split functions to another file(`src/dbt_osmosis/core/column_level_knowledge_propagator.py`)
  - Defined types for `ColumnLevelKnowledge` and `Knowledge` to understand code easier
- Adding some tests
  - In this pull request, we are only adding functions for which tests are relatively easy to add (to avoid making the differences in this pull request too large and difficult to review)
  - We intend to add more detailed tests and tests for other functions in subsequent pull requests
- Run pytest on GitHub Actions

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,11 +39,6 @@ dbt-duckdb = { version = ">=1.0.0", optional = true }
 dbt-sqlite = { version = ">=1.0.0", optional = true }
 dbt-postgres = { version = ">=1.0.0", optional = true }
 
-[tool.poetry.dev-dependencies]
```
Comment (syou6162) [inline pyproject.toml:42]: `dev-dependencies` is deprecated.

- ref: https://python-poetry.org/docs/managing-dependencies/

File: tests/data/manifest.json [hunk 2]
```diff
@@ -0,0 +1,11347 @@
+{
```
Comment (syou6162) [inline tests/data/manifest.json:1]: I made this file with `dbt build --target test` in `demo_duckdb` directory.

File: src/dbt_osmosis/core/column_level_knowledge_propagator.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    ManifestNode,
+)
+
+
+ColumnLevelKnowledge = Dict[str, Any]
+Knowledge = Dict[str, ColumnLevelKnowledge]
+
+
+def _build_node_ancestor_tree(
+    manifest: ManifestNode,
```
Comment (syou6162) [inline src/dbt_osmosis/core/column_level_knowledge_propagator.py:18]: Fixed to take `manifest` as a function argument to make testing easier

**Review by z3z1ma — APPROVED** [review]
Very nice improvements, clean code. 💯 

LGTM

**Pull request merged**